### PR TITLE
Added python test runner (nose) and a yaml linter to travis

### PR DESCRIPTION
**Pull request by Br3nda**
Also fixed yaml files.

### CODE REVIEW

**Comment by Br3nda** [comment]
if you use unittest.TestCase for the assertions, it'll give more info what is different (rather than just a failure)

e.g.

```python
import os
import taas
import unittest


class TestTaas(unittest.TestCase):

    def setUp(self):
        # Clean our environment of variables that may screw our tests
        if os.environ.get('TAS_DATA', None):
            del os.environ["TAAS_DATA"]
        self.test_root = os.path.dirname(__file__)
        self.test_config = os.path.join(self.test_root, "config.yml")

    def test_read_config(self):
        config = taas.read_config(self.test_config)

        self.assertNotEqual(config['sources'], None)
        self.assertNotEqual(config['sources']['bad_key'], None)

    def test_taas_data_env(self):
        """
        Tests that we handle the TAAS_DATA environment correctly.
        """

        # Clear env first, just in case

        fake_env = "/tmp/somefakedir"

        data = taas.data_root()
        self.assertNotEqual(data, None)

        os.environ["TAAS_DATA"] = fake_env
        data_env = taas.data_root()

        self.assertNotEqual(data_env, data)
        self.assertEqual(data_env, fake_env)

    def test_data_roots(self):
        """
            Tests our data roots in various ways.
        """

        data_root = taas.data_root()

        self.assertNotEqual(taas.data_root(), None)

        self.assertEqual(taas.sheets_root(), os.path.join(data_root, "sheets"))
        self.assertEqual(taas.json_root(), os.path.join(data_root, "json"))
```

**Comment by pjf** [comment]
@Br3nda : You so, *sooooo* rock! 

I was wondering the best way to do testing; I'm used to `Test::More` and friends giving really helpful output on failures, but wasn't sure how that was done in Python yet. Now I am!

I appreciate your contributions and feedback *so* friggin' much. And thank you for the Code Climate link, I hadn't used them before!

File: .travis.yml [hunk 1]
```diff
@@ -1,14 +1,22 @@
...
 language: python
 
 python:
-    - "2.7"
+  - 2.7
 
-install: "make"
+install:
+  - pip install -r requirements.txt
+  - pip install yamllint pep8
```
Comment (pjf) [inline .travis.yml:9]: Is this the way we should split build/run requirements from test/authortest requirements?

**Comment by pjf** [comment]
Merging with great, great thanks! You're amazing! <3

File: .travis.yml (hunk 1 above)
Comment (Br3nda) [reply .travis.yml:9]: probably not - there's possibly a smarter way to specify dev dependencies in requirements.txt

**Pull request merged**